src/core/highlight-vars.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Module core/highlight-vars
+ * Highlights occurrences of a <var> within a section on click.
+ * Set `conf.highlightVars = true` to enable.
+ * Removes highlights from <var> if clicked anywhere else.
+ * All is done while keeping in mind that exported html stays clean
+ * on export.
+ */
+import { sub } from "core/pubsubhub";
+import hlVars from "deps/text!core/css/var.css";
+
+export const name = "core/highlight-vars";
+
+export function run(conf) {
+  if (!conf.highlightVars) {
+    return;
+  }
+  const styleElement = document.createElement("style");
+  styleElement.textContent = hlVars;
+  styleElement.classList.add("removeOnSave");
+  document.head.appendChild(styleElement);
+
+  [...document.querySelectorAll("var")].forEach(varElem =>
+    varElem.addEventListener("click", hightlightListener)
+  );
+
+  // remove highlights, cleanup empty class/style attributes
+  sub("beforesave", outputDoc => {
+    [...outputDoc.querySelectorAll("var.respec-active")].forEach(
+      removeHighlight
+    );
+  });
+}
+
+function hightlightListener(ev) {
```

Comment:
Did a little refactor here. Instead of querying the body for highlighted elements, we just unhighlight the ones that were highlighted.